L79-cpp-async-response-reader-destruction.md

```diff
@@ -0,0 +1,123 @@
+API changes for `ClientAsyncResponseReaderInterface` destruction
+----
+* Author: jacobsa
+* Approver: ctiller
+* Status: Draft
+* Implemented in: C++
+* Last updated: 2021-05-14
+* Discussion at: <google group thread> (filled after thread exists)
+
+## Abstract
+
+Currently although
+[`grpc::ClientAsyncResponseReaderInterface`](https://github.com/grpc/grpc/blob/7e14d23ab46e1b0924d6c3e40797ed1d587aee7a/include/grpcpp/impl/codegen/async_unary_call.h#L37-L72)
+has a virtual destructor, `std::default_delete` is specialized such that the
+destructor never actually runs. So the only subclasses that can actually be
+implemeneted without hacks are those like `grpc::ClientAsyncResponseReader`,
+which knows it is always allocated on an arena and doesn't care about its
+destructor running. This proposal provides a plan for fixing this so that other
+subclasses can count on their destructor running and their backing memory being
+freed, as with any other C++ class.
```

Comment:
>This proposal provides a plan for fixing this so that other
> subclasses can count on their destructor running and their backing memory being
> freed, as with any other C++ class.

I would replace this with: 

```
This proposal provides a plan for guaranteed destructor invocation, such that 
subclasses can count on their destructor running and their backing memory being
freed, as with any other C++ class.
```

Reply:
Done.